sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract features across (shifted) windows from provided series."""
+
+__author__ = ["Daniel Bartling"]
+__all__ = ["LaggedWindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+# from sktime.transformations.base import _PanelToTabularTransformer
+from sktime.transformations.base import BaseTransformer
+
+# from sktime.utils.validation.panel import check_X
+# List of native pandas rolling window function.
+# danbartl: investivate different engines for pandas
+pd_rolling = [
+    "sum",
+    "mean",
+    "median",
+    "std",
+    "var",
+    "kurt",
+    "min",
+    "max",
+    "corr",
+    "cov",
+    "skew",
+    "sem",
+]
+
+
+def _window_feature(
+    Z,
+    name=None,
+    win_summarizer=None,
+    window=None,
+):
+    """Compute window features and lag.
+
+    y: either pandas.core.groupby.generic.SeriesGroupBy
+        Object create by grouping across groupBy columns.
+    name : str, base string of the derived features
+           final name will also contain
+           window shift and window length.
+    win_summarizer: either str corresponding to native
+    implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+         or function definition
+    window: list of integers
```

Comment:
you need to write more detail here, or it will be hard to understand what it does.
Use math notation or array notation.